tests/test_lint_maturity.py

```diff
@@ -0,0 +1,118 @@
+2020.02.b1-py27.yml has 5 packages with invalid maturity tag.
+\tTag a packages: {'package_a1': 'v3.1.a1', 'package_a2': '3.1.a2'}
+\tTag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+2020.02.rc1-py27.yml has 7 packages with invalid maturity tag.
+\tTag a packages: {'package_a1': 'v3.1.a1', 'package_a2': '3.1.a2'}
+\tTag b packages: {'package_b1': 'v3.1.b1', 'package_b2': '3.1.b1+ert'}
+\tTag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+bleeding-py27.yml has 9 packages with invalid maturity tag.
+\tTag a packages: {'package_a1': 'v3.1.a1', 'package_a2': '3.1.a2'}
+\tTag b packages: {'package_b1': 'v3.1.b1', 'package_b2': '3.1.b1+ert'}
+\tTag rc packages: {'package_rc1': 'v3.1.rc1', 'package_rc2': '3.1.rc1'}
+\tTag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+"""
+
+EXPECTED_WARNING = """UserWarning: 2020.02.01-py27.yml, exception list of packages:
+\tpackage_IV2 
+2020.02.a1-py27.yml, exception list of packages:
+\tpackage_IV2 
+2020.02.b1-py27.yml, exception list of packages:
+\tpackage_IV2 
+2020.02.rc1-py27.yml, exception list of packages:
+\tpackage_IV2 
+bleeding-py27.yml is in the exception list of releases. 
+bleeding-py27.yml, exception list of packages:
+\tpackage_IV2 
+  
+warnings.warn(system_warning_msg, UserWarning)"""
+
+
+TAG_EXCEPTIONS = {"release": ["bleeding", "rpath"], "package": ["package_IV2"]}
+
+
+def create_test_files(release_sample, file_names_sample):
+    folder_name = os.getcwd() + "/test_lint_maturity/"
+    os.mkdir(folder_name)
+    list_files = []
+    for file_name in file_names_sample:
+        list_files.append(folder_name + file_name)
+        with open(folder_name + file_name, "w") as file_sample:
+            file_sample.write(release_sample)
+    return list_files
+
+
+def test_lint_maturity_files(tmpdir):  # tmpdir
+    with tmpdir.as_cwd():
+        list_files = create_test_files(RELEASE_SAMPLE, FILE_NAMES_SAMPLE)
+    with pytest.raises(SystemExit) as exit_info, warnings.catch_warnings(
+        record=True
+    ) as warning_info:
+        lint_maturity.run(list_files, TAG_EXCEPTIONS)
+    assert str(warning_info[0].message) in EXPECTED_WARNING
+    assert EXPECTED_SYSTEMEXIT in str(exit_info.value)
+
+
+def test_lint_maturity_pass(tmpdir):
+
+    with tmpdir.as_cwd():
+        file_name = create_test_files(
+            RELEASE_SAMPLE.rsplit("\n", 4)[0], [FILE_NAMES_SAMPLE[1]]
+        )
+    try:
+        with pytest.raises(SystemExit) as exit_info, warnings.catch_warnings(
+            record=True
+        ) as warning_info:
+            lint_maturity.run(file_name, TAG_EXCEPTIONS)
+        pytest.fail("SystemExit was raised when it should not.")
+    except:  # check
+        pass
```

Comment:
This is a bit confusing to me, because it looks like you want it to raise an exception, when in fact you´re doing the excact opposite, you could do something like this:
```suggestion
    with pytest.warns(UserWarning) as record:
        lint_maturity.run(file_name, TAG_EXCEPTIONS)

    assert record[0].message.args[0] == "2020.02.a1-py27.yml, exception list of packages:\n"
```

Though looking at it, you might not want it to raise that warning at all in this case?

Reply:
If you don´t want it to raise any warnings you could confirm with:
```suggestion
def test_lint_maturity_pass(tmpdir):

    with tmpdir.as_cwd():
        file_name = create_test_files(
            RELEASE_SAMPLE.rsplit("\n", 4)[0], [FILE_NAMES_SAMPLE[1]]
        )
    with pytest.warns(None) as record:
        lint_maturity.run(file_name, TAG_EXCEPTIONS)

    assert not record
```

Reply:
Thanks Øyvind! I understand this may be confusing. 
Here I would like to test that nothing is raised (warning and/or SystemExit) when all packages have appropriate maturity for a giving release. 
It seems that the suggested code would not test this, but perhaps renaming would be a way to improve and avoid confusion. 
Would you have a better suggestion for the name test_lint_maturity_pass to represent this objective?

Reply:
That´s what I thought after thinking some, my second suggestion will assert that, but will currently fail as a warning was raised when I ran your code.

Reply:
And it should be enough to assert that no warnings are raised, as a `SystemExit` will cause the test to fail.